src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentServer.kt

```diff
@@ -92,4 +92,9 @@ interface CodyAgentServer {
 
   @JsonRequest("chat/submitMessage")
   fun chatSubmitMessage(params: ChatSubmitMessageParams): CompletableFuture<ExtensionMessage>
+
+  @JsonRequest("chat/models")
+  fun chatModels(params: ChatModelsParams): CompletableFuture<ChatModelsResponse>
+
+  @JsonRequest("chat/restore") fun chatRestore(params: ChatRestoreParams): CompletableFuture<String>
```

Comment:
Can we use a type alias here or add some comment? We have already discussed that panelid/sessionid/etc, but when I'm looking on this code with fresh eyes, it's still not super clear what's in the response (since the type is `String`).

Reply:
I'm not 100% sure if I'm free to do it in the serialisable protocol classes, but I can give it a try.

Reply:
I'm 99% sure it will work. It's just an alias that you can later mix with a string and the compiler shouldn't complain.

Reply:
I'm deferring this for later. `ChatRestoreParams` needs to be also improved on the agent side, I plan to do it a bit later.